http/src/lib.rs

```diff
@@ -503,8 +504,9 @@ fn serve<M: jsonrpc::Metadata, S: jsonrpc::Middleware<M>>(
 
 			let mut http = server::conn::Http::new();
 			http.keep_alive(keep_alive);
+			let tcp_stream = SuspendableStream::new(listener.incoming(), Duration::from_secs(2));
```

Comment:
2 seconds might be a bit too much. What do you think about making the delay exponential? I.e. with every delay followed by another failed attempt we multiply the delay by 2. Thanks to this we could start with some really low values (like 5-10ms and increase dynamically if there are no open files available).